### ISSUE TITLE
LeftBar/RightBar UI block error

### ISSUE THREAD

**Issue by cloudrco** [grahammendick/navigation#502]
Getting an exception error while having an Image component as a child, for both LeftBar and RightBar.

<img width="325" alt="Screen Shot 2021-01-16 at 10 47 42 AM" src="https://user-images.githubusercontent.com/64134460/104791959-5e3cbc80-57e8-11eb-81b0-8b818bb0d638.png">

```
<LeftBar>
  <Image source={{uri: "https://source.unsplash.com/-djRG1vB1pw/100x100"}} style={{...}} />
</LeftBar>
```

I tried using BarButton and works well, but displays a blue box instead of the actual image.
```
<LeftBar>
  <BarButton image={{uri: "https://source.unsplash.com/-djRG1vB1pw/100x100"}} />
</LeftBar>
```
<img width="324" alt="Screen Shot 2021-01-16 at 10 51 55 AM" src="https://user-images.githubusercontent.com/64134460/104792072-e1f6a900-57e8-11eb-9e0e-504f56957b7a.png">

The TitleBar however, works as expected.
```
<TitleBar>
  <Image
    source={{uri: "https://source.unsplash.com/-djRG1vB1pw/100x100"}}
    style={{...}}
  />
</TitleBar>
```
<img width="324" alt="Screen Shot 2021-01-16 at 10 57 54 AM" src="https://user-images.githubusercontent.com/64134460/104792400-0737e700-57ea-11eb-8b0b-6f6d0c92726a.png">

**Comment by grahammendick** [comment]
You were right to use `BarButton`. For example, here's [a uri image](https://www.gstatic.com/images/icons/material/system/1x/grade_white_20dp.png) added to the Details scene of the [zoom sample](https://github.com/grahammendick/navigation/tree/master/NavigationReactNative/sample/zoom)
```jsx
<RightBar>
  <BarButton image={{uri: "https://www.gstatic.com/images/icons/material/system/1x/grade_white_20dp.png"}} onPress={() => {
    stateNavigator.navigateBack(1);
  }} />
</RightBar>
```
And here's what it looks like

![Screenshot 2021-01-16 at 10 51 06](https://user-images.githubusercontent.com/1761227/104809945-22562700-57e9-11eb-89fb-c66cafdbe197.png)

You see, iOS colors anything that isn't transparent in the image to match the 'tintColor' of the navigation bar. For example, If I keep the same image and set the 'tintColor' prop to 'red' then it looks like

![Screenshot 2021-01-16 at 10 55 02](https://user-images.githubusercontent.com/1761227/104809984-88db4500-57e9-11eb-9669-6c2ffb33fda9.png)

Take a look at some of the native apps in your phone, like WhatsApp and Twitter, and you'll see that this is how they all behave. It's the same on iOS and Android. The bar button images are all single color icons. It's the same for tab bar images too which explains [the limited customization options](https://github.com/grahammendick/navigation/issues/501#issuecomment-761072996). 

The Navigation router gives you an app that looks and feels like other native apps because all its components, including the `BarButton`, render to the underlying native behaviour.

**Comment by cloudrco** [comment]
Thanks @grahammendick. Tried using react-native-screens (native stack) with react navigation and this is very accomplishable. https://github.com/software-mansion/react-native-screens/tree/master/native-stack#headerleft

<img width="324" alt="Screen Shot 2021-01-17 at 10 28 08 AM" src="https://user-images.githubusercontent.com/64134460/104827907-0cab3500-58af-11eb-9658-6f88a80e15d0.png">

**Comment by grahammendick** [comment]
Yes, it's possible using the [customView on a UIBarButtonItem](https://developer.apple.com/documentation/uikit/uibarbuttonitem/1617129-customview?language=objc). Would you be interested in working on a PR that adds custom view support to the `BarButton` component?

**Comment by grahammendick** [comment]
If you don't want to work on a PR that's no problem, I'm happy to work on it. Just thought it would be a good chance for you to see how React Native and iOS native fit together. I'll help you as much as you need if you're interested

**Comment by cloudrco** [comment]
Hi @grahammendick, sorry was caught up with work. I'll be interested on this PR, however I don't know how long or when I'll ever manage to submit it. I have little to no experience on the native side of things 😔. If you are willing to do so, that'll be awesome 😃. Thanks!

**Comment by grahammendick** [comment]
No problem, I'll close this because it's a duplicate of #496. Thanks for raising this because I was hoping for more use-cases to convince me it was needed. If you change your mind about the PR then let me know

**Comment by grahammendick** [comment]
@cloudrco I just [released navigation-react-native 7.4.0](https://github.com/grahammendick/navigation/releases/tag/v7.4.0-NavigationReactNative) with custom `BarButton` views on iOS

**Comment by grahammendick** [comment]
For example,

```jsx
<LeftBar>
  <BarButton>
    <TouchableHighlight onPress={() => {}}>
      <Image />
    </TouchableHighlight>
  </BarButton>
</LeftBar>
```

**Comment by dbarcinas** [comment]
Hello @grahammendick, thanks for this! Exactly what we needed. Kudos! 😃

**Comment by grahammendick** [comment]
@dbarcinas Hey, that's great to hear!

**Comment by cloudrco** [comment]
Thank you @grahammendick. Much appreciated 👍

**Comment by grahammendick** [comment]
@cloudrco You're welcome 👍

**Comment by alexstanbury** [comment]
Hi Graham, is this feature also available on Android? My use case is I want to have my own SVG icon on the right side of the nav bar that when triggered will open a drawer, so far I can only see how to use the native Android 3 dots icon that then opens and floating menu, which is not what I need to do.

**Comment by grahammendick** [comment]
Hey @alexstanbury, nice to hear from you!

On Android, if you set the `show` prop to `always` then the image button will appear on the right hand side of the navigation bar
```jsx
<RightBar>
  <BarButton
    show=”always”
    image={require('./yourimage.png')}
    onPress={() => setDrawerOpen(true)} />
</RightBar>
```

**Comment by alexstanbury** [comment]
Thanks, that's got me part of the way there, but using this approach I don't think it would be possible to animate say a hamburger icon into a cross icon when the drawer is opened, would it? Also, are there API docs available anywhere or should I just stick to browsing the source?

**Comment by grahammendick** [comment]
I haven't found a way to get `BarButton` children working on Android.

For API docs, I recommend using the [typescript definitions](https://github.com/grahammendick/navigation/blob/master/build/npm/navigation-react-native/navigation.react.native.d.ts) that ship with the package. Please let me know if these help

**Comment by alexstanbury** [comment]
Ah that's a shame, I should be able to make do with just a static image. Thanks for the quick replies 👍

**Comment by grahammendick** [comment]
@alexstanbury I know how to get `BarButton` children working on Android. Let me know if you're interested in working on the PR and I'll fill you in on the details

**Comment by grahammendick** [comment]
@alexstanbury Thanks, that's great. Whenever you're free just let me know and we can go through it

**Comment by alexstanbury** [comment]
Hi, I'd love to have the time to be able to but I'm absolutely stacked for
the foreseeable future 😞 sorry man.

**Comment by grahammendick** [comment]
@alexstanbury I just released [navigation-react-native 8.7.0](https://github.com/grahammendick/navigation/releases/tag/v8.7.0-NavigationReactNative) with custom `BarButton` views on Android (they’re already available on iOS). For example,
```jsx
<RightBar>
  <BarButton show="always">
    <TouchableHighlight onPress={() => {}}>
      <Image />
    </TouchableHighlight>
  </BarButton>
</RightBar>
```

**Comment by alexstanbury** [comment]
@grahammendick Lovely! I'll have a play around with it ASAP, cheers!
